src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -124,6 +124,16 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("aa+", "a{2,}")]
         [InlineData("aa?", "a{1,2}")]
         [InlineData("aa{1,3}", "a{2,4}")]
+        [InlineData("aaa+b", "aa(?>a+)b")]
+        [InlineData("a+aab", "(?>a{3,})b")]
+        [InlineData("a{3}b", "aaab")]
```

Comment:
This test only verifies that they both reduce the same. Your change avoids reducing `aaa` into `a{3}` but this is only testing that they're equivalent. Is that useful ? -- I imagine 1000 tests would fail if it wasn't true.

Reply:
> Your change avoids reducing aaa into a{3} but this is only testing that they're equivalent

This is testing that they reduce to the same thing, i.e. that the node trees they produce are identical.  That's not the same as their having the same execution semantics.

Reply:
Right, but it would pass whether the node tree produced had a loop or had a multi. And this change is designed to ensure it ends up as the right one. Both trees will also produce the same results for the user. The test is obviously worth having -- just wondering aloud about whether we need a new kind of test that verifies some aspect of the node tree.

Reply:
> Right, but it would pass whether the node tree produced had a loop or had a multi

Yes, it's not verifying the direction it takes, just that they reduce to the same thing.  If that's what you were saying, I'm not sure what you meant about "I imagine 1000 tests would fail if it wasn't true".

Reply:
Sorry, my communication is poor. I figure there's 3 aspects

* does expression X1 reduce to the same node tree as expression X2 -- these tests.
* is that node tree structured as ABC, or have property D 
* does expression X functionally behave as expected --- this is what we have 1000's of tests for.

I'm wondering whether, if I reversed this PR so that multi was collapsed into a loop or whatnot, we'd see test failures, because we do'nt have the 2nd category

Reply:
Eg., if the node tree was serialized into a descriptive string (debug only) then a test could verify that expression X formed serialized node tree S.

Reply:
These expressions behave identically and reduce to the same thing; the only thing that would be different based on which direction they reduced would be their execution performance.  We do not currently have tests that validate the tree it exactly the set of nodes we expect it to be, just these that validate two expressions parse into the same exact tree.

Reply:
right, agreed.